tests/test_http_client.py

```diff
@@ -1505,3 +1524,418 @@ def test_timeout(self):
 
     def test_timeout_async(self):
         pass
+
+
+class TestAIOHTTPClient(StripeClientTestCase, ClientTestBase):
+    REQUEST_CLIENT: Type[
+        _http_client.AIOHTTPClient
+    ] = _http_client.AIOHTTPClient
+
+    @pytest.fixture
+    def mock_response(self, mocker, request_mock):
+        def mock_response(mock, body={}, code=200):
+            class Content:
+                def __aiter__(self):
+                    async def chunk():
+                        yield bytes(body, "utf-8") if isinstance(
+                            body, str
+                        ) else body
+
+                    return chunk()
+
+                async def read(self):
+                    return body
+
+            class Result:
+                def __init__(self):
+                    self.content = Content()
+                    self.status = code
+                    self.headers = {}
+
+            result = Result()
+
+            request_mock.ClientSession().request = AsyncMock(
```

Comment:
Is it expected that while testing HTTP client we are not making actual requests using the HTTP client?

Reply:
I was under impression that we are starting an in-process http server.

Reply:
Yes `test_integration` makes end-to-end requests and I have expanded it in this PR. `test_http_client` mocks the underlying http client implementation but exercises the code in the http client "wrapper".

Reply:
My take is that the integration tests are way more useful, but I've cargo-culted these and extended them to the new http clients, mostly begrudgingly, 80% of the time when these fail it's something dumb with the mock instead of a real problem with the implementation, but I wouldn't say they're completely useless.